### ISSUE TITLE
Export  torch script with dynamic batch , nms  and FP16

### ISSUE THREAD

**Issue by 631068264** [ultralytics/ultralytics#19741]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

I want to get this model for Triton Server with pytorch backend

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @631068264, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for guidance, where you can find detailed examples for [exporting models](https://docs.ultralytics.com/modes/export/) and other use cases.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/), including the specific code and commands you used, to help us debug the issue.

If this is a feature ❓ Question, please provide additional details about your requirements, such as specific configurations or challenges you're facing when exporting the model for Triton Server with PyTorch backend. This will help us understand your use case better!

In the meantime, ensure you are running the latest version of the `ultralytics` package. You can upgrade with:

```bash
pip install -U ultralytics
```

## Export Support

YOLO models can be exported to various formats for deployment. Check out the [export guide](https://docs.ultralytics.com/modes/export/) for supported formats and options, including TorchScript and FP16. For dynamic batching and NMS configurations, you may need to customize the export process—feel free to share specifics if you encounter challenges.

## Community and Support

Join the Ultralytics community to explore and discuss your use case further:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for community interactions

## Verified Environments

Verify your setup using one of these environments with all dependencies preinstalled:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

An Ultralytics engineer will assist you further soon. Thanks for being part of the community 🌟!

**Comment by 631068264** [comment]
@UltralyticsAssistant Not work

**Comment by 631068264** [comment]
If I use `torch.jit.script` to  export such as `task.py  , head.py , module.py` will get error  and I don't know how to fix them.

And I use it after `torch.jit.trace` get same error when use different batch size to inference
```
   @try_export
    def export_torchscript(self, prefix=colorstr("TorchScript:")):
        """YOLO TorchScript model export."""
        LOGGER.info(f"\n{prefix} starting export with torch {torch.__version__}...")
        f = self.file.with_suffix(".torchscript")

        ts = torch.jit.trace(NMSModel(self.model, self.args) if self.args.nms else self.model, self.im, strict=False)
        ts = torch.jit.script(ts)
        extra_files = {"config.txt": json.dumps(self.metadata)}  # torch._C.ExtraFilesMap()
        if self.args.optimize:  # https://pytorch.org/tutorials/recipes/mobile_interpreter.html
            LOGGER.info(f"{prefix} optimizing for mobile...")
            from torch.utils.mobile_optimizer import optimize_for_mobile

            optimize_for_mobile(ts)._save_for_lite_interpreter(str(f), _extra_files=extra_files)
        else:
            ts.save(str(f), _extra_files=extra_files)
        return f, None

```


I don't know how to solve my problem

**Comment by glenn-jocher** [comment]
@631068264 to export a YOLO model to TorchScript with dynamic batch, NMS, and FP16, use the official export arguments without manual scripting:

```python
model.export(format='torchscript', dynamic=True, nms=True, half=True)
```

This handles dynamic batching and NMS automatically. The error suggests dimension mismatches - ensure your model is compatible with dynamic axes (batch dimension should be first). For Triton deployment, consider ONNX or TensorRT formats which often work better with dynamic shapes. If issues persist, please share a minimum reproducible example using the latest Ultralytics version.

**Comment by 631068264** [comment]
@glenn-jocher   So this is a bug   https://github.com/ultralytics/ultralytics/issues/19745

**Comment by glenn-jocher** [comment]
@631068264 thanks for flagging this! We'll review the TorchScript export issue in [#19745](https://github.com/ultralytics/ultralytics/issues/19745). For immediate solutions, we recommend using the official export method with `model.export(format='torchscript', dynamic=True, nms=True, half=True)` which handles dynamic axes and NMS integration automatically. Ensure you're on the latest Ultralytics version. If issues persist, please share a minimum reproducible example as shown in our [Bug Report Guide](https://docs.ultralytics.com/help/contributing/#minimum-reproducible-example).
